lib/rspotify/user.rb

```diff
@@ -118,6 +118,7 @@ def create_playlist!(name, public: true)
     def currently_playing
       url = "me/player/currently-playing"
       response = RSpotify.resolve_auth_request(@id, url)
+      return nil unless response && response['item']
```

Comment:
What do you think about using [fetch](https://ruby-doc.org/core-2.2.0/Hash.html#method-i-fetch) to accomplish this?

Reply:
Very new to Ruby so wasn't aware of this, looks interesting though so will take a look at adding it :)

Reply:
Part of the problem seems to be that response itself is coming back as nil, so I can't do response.fetch because that also throws a undefined method for nil error.

Reply:
Ah, of course, I should have known that. It would be nice to examine the code paths that are returning `nil` here in the first place, since that is the real root of this evil.

Reply:
I just dug a little bit. This is just my personal opinion, it may not actually be useful, but I would think a nice way to handle this would be to return some kind of [null object](https://en.wikipedia.org/wiki/Null_object_pattern) from [here](https://github.com/guilhermesad/rspotify/blob/c30723d87559e1ae5272a85e5399be8411347937/lib/rspotify/connection.rb#L75) when the response is nil or empty.

Reply:
Yup, agree completely. Looks like the send_request is used by all (or most) of the other methods so that will fix this problem across the board rather than just for the currently playing track. 

I've also found a bug with character encoding that I needed to change the send_request method for, so I may include that in a future pull request as well.

Reply:
Just realising a slight problem with that, surely returning a null object from send_request is still going to require a check in currently_playing (or any other method in user) to determine if the response has a value or is a null object, which would essentially be the same as what the nil check is doing? Otherwise at some point Json is going to try parse whatever response is, and if it isn't able to that's going to cause problems.

Reply:
Totally, but checking for a specific class is better than checking for nil I think. 🤷‍♂️ ruby problems.

Reply:
We could also have a NullObject that returns empty collections for the things it is supposed to have, if that makes sense. So `response[item]` could return `''` or something like that.